src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,85 @@
+package duke.task;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * Represents a task in the Duke application.
+ * This abstract class serves as the base for different types of tasks that can be created.
+ */
+public abstract class Task {
+
+    private String task;
+    protected boolean isDone;
+
+    /**
+     * Constructs a task with the given task description.
+     *
+     * @param task The description of the task.
+     */
+    public Task(String task) {
+        this.task = task;
+        this.isDone = false;
+    }
+
+    /**
+     * Constructs a task with the given task description and completion status.
+     *
+     * @param task The description of the task.
+     * @param isDone The completion status of the task.
+     */
+    public Task(String task, boolean isDone) {
+        this.task = task;
+        this.isDone = isDone;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : "  ");
+    }
+
+    /**
+     * Returns the string representation of the task.
+     *
+     * @return A string containing the task's status icon and description.
+     */
+    @Override
+    public String toString() {
+        return "[" + this.getStatusIcon() + "] " + task;
+    }
+
+    /**
+     * Marks the task as completed.
+     *
+     * @return The completed task.
+     */
+    public Task completeTask() {
```

Comment:
Perhaps a better name for this method would be `doTask()` ? So that it may comply with the `undoTask()` method